### ISSUE TITLE
WinRT: Windows 8.1 SafeFireAndForget issue "No overload for method"

### ISSUE THREAD

**Issue by JeromeGsq** [brminnick/AsyncAwaitBestPractices#33]
Hello,

I noticed an error on my WinRT project (Windows 8.1) using your package.
I can't use the SafeFireAndForget extension method because I have this message :
`No overload for method 'SafeFireAndForget' takes 1 arguments`

![image](https://user-images.githubusercontent.com/6270529/78156109-96803380-743e-11ea-8cc1-36c4f950f2ae.png)

Do you have more information about this problem?

Thanks

Nuget package version : 4.1.0

Microsoft Visual Studio Enterprise 2015
Version 14.0.25431.01 Update 3
Microsoft .NET Framework
Version 4.8.03752

**Comment by brminnick** [comment]
Hi @jeromegsq!

Change it to this:
`.SafeFireAndForget(ex => Debug.WriteLine(ex))`

**Comment by JeromeGsq** [comment]
@brminnick 
Thank you but even with this, I have the same message : 
![image](https://user-images.githubusercontent.com/6270529/78157600-6fc2fc80-7440-11ea-99c2-23205a85ed3e.png)

**Comment by brminnick** [comment]
Strange - What happens if you do this?

`.SafeFireAndForget(ex => Debug.WriteLine(ex), false)`

**Comment by JeromeGsq** [comment]
Doesn't work too :

**Comment by brminnick** [comment]
Can you upload a reproduction sample so that I can dive into this problem?

I can’t reproduce it on my end.

**Comment by JeromeGsq** [comment]
I will try to send you a sample soon. 
Thank you for your help

**Comment by JeromeGsq** [comment]
[WpfApplication1.zip](https://github.com/brminnick/AsyncAwaitBestPractices/files/4416389/WpfApplication1.zip)
Here is a WPF sample with Visual Studio 2015.

**Comment by brminnick** [comment]
Thanks!

Confirmed on Visual Studio 2015 Update 3 👍

**Comment by brminnick** [comment]
Confirmed no issues on Visual Studio 2019 v16.5.2

**Comment by brminnick** [comment]
Looks like Visual Studio 2015 requires you to create a variable for `ref` arguments.

Here's a workaround:

```csharp
void Workaround()
{
    Action<Exception> handleException = ex => Debug.WriteLine(ex);
    bool continueOnCapturedContext = false;

    Run().SafeFireAndForget<Exception>(ref handleException, ref continueOnCapturedContext);
}

async Task Run()
{
    await Task.Delay(2000);
}
```

**Comment by brminnick** [comment]
The fix is now available in v4.1.1-pre1 🎉 

@JeromeGsq When you get a chance, can you try out v4.1.1-pre1 and confirm it works for you?

NuGet: https://www.nuget.org/packages/AsyncAwaitBestPractices/4.1.1-pre1
GitHub Release: https://github.com/brminnick/AsyncAwaitBestPractices/releases/tag/v4.1.1-pre1

```csharp
void FixedIn411Pre1()
{
    Run().SafeFireAndForget(ex => Debug.WriteLine(ex));
}

async Task Run()
{
    await Task.Delay(2000);
}
```

**Comment by JeromeGsq** [comment]
@brminnick 
Thank you very much, I confirm that the problem is fixed with this version! 
 🎉
